File: old/unix-shopping.py
import sys
sys.setrecursionlimit(1000000000)

input_file = open("shopin.txt", "r")
output_file = open("shopout.txt", "w")

input_line = input_file.readline().strip()
N, K = map(int, input_line.split())
input_line = input_file.readline().strip()
costs = list(map(int, input_line.split()))
totalCost = 0

for i in range(K):
    if len(costs) < 2:
        break
    cheapest = costs[0]
    print(cheapest)
    costs = costs[1:-1]
    totalCost += cheapest

for i in range(len(costs)//2):
    if len(costs) < 2:
        break
    print(totalCost)
    totalCost += costs[-1]
    costs = costs[:-2]

answer = totalCost

output_file.write("%d\n" % (answer))

input_file.close()
output_file.close()

File: old/unix-shopping.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>

int main() {
    std::ifstream input_file("shopin.txt");
    std::ofstream output_file("shopout.txt");

    int N, K;
    input_file >> N >> K;

    std::vector<int> costs(N);
    for (int i = 0; i < N; ++i) {
        input_file >> costs[i];
    }

    int totalCost = 0;

    for (int i = 0; i < K; i++) {
        if (costs.size() < 2) {
            break;
        }
        int cheapest = costs[0];
        costs.erase(costs.begin());
        costs.pop_back();
        totalCost += cheapest;
    }

    for (int i = 0; i < costs.size() / 2; i++) {
        if (costs.size() < 2) {
            break;
        }
        totalCost += costs.back();
        costs.pop_back();
        costs.pop_back();
    }

    output_file << totalCost << std::endl;

    input_file.close();
    output_file.close();

    return 0;
}